### ISSUE TITLE
loginWithPopup can fail but still allow the social account to be created in auth0 database

### ISSUE THREAD

**Issue by qortex** [auth0/auth0-spa-js#513]
### Describe the problem

For timeout (and/or other failures?), `loginWithPopup` can return a failure, whereas the flow actually succeeds in the login popup. Thus, the underlying app has wrong informations about the login flow, and it leads to all sorts of wrong behaviours.

### What was the expected behavior?

Return value from `loginWithPopup` should **always** be in sync with the actual success.

### Reproduction

1. Initiate a social connection with `loginWithPopup`
2. Let it timeout -> the promise rejects so your app thinks it failed
3. But in the open window, complete the flow after the timeout
4. Now you are in a wrong state: the app thinks the login failed, but the account has been created in auth0 database. Thus, blocking issues for the flow afterwards (all kinds of assumptions fail in the app state).

### Environment

- **Version of `auth0-spa-js` used**: 1.10.0
- **Which browsers have you tested in?**: Chrome
- **Which framework are you using, if applicable (Angular, React, etc)**: Angular
- **Other modules/plugins/libraries that might be involved**: none

**Comment by qortex** [comment]
Workaround for now, I improved a lot the timeout: 
```
client.loginWithPopup({ connection: name }, { timeoutInSeconds: 600 })
```

**Comment by stevehobbsdev** [comment]
Thanks for raising. This needs investigation but sometimes we're at the whim of how Auth0 server reports errors to us. Let us have a look.

As a reminder, the issue template that you use has all sorts of useful information on it for a reason - SDK version, environment, and other questions. Can you please provide that info?

**Comment by qortex** [comment]
Yeah right, sorry about that, edited the initial ticket (all latest versions).

**Comment by stevehobbsdev** [comment]
Sorry for the delay here @qortex. We think that #422 may fix this. As it's had no progress for a while, we're planning to take that over and complete the work. Glad the workaround is working for you in the meantime.

**Comment by stevehobbsdev** [comment]
👋 We released #422 into v1.13.1, which might make a difference.

Otherwise, specifying a longer timeout here as you have done is the mitigation that we recommend if you're running into these issues. We rely a lot on using `postMessage` with these kinds of interactions with Auth0 Server and if it doesn't call back to you in time, we're limited as to what can happen other than just refreshing the page.

Closing for now, but happy to discuss further.
